### PR TITLE
[Merged by Bors] - style: use `cases x with | ...` instead of `cases x; case => ...`

### PR DESCRIPTION
**Pull request by digama0**
This converts usages of the pattern
```lean
cases h
case inl h' => ...
case inr h' => ...
```
which derive from mathported code, to the "structured `cases`" syntax:
```lean
cases h with
| inl h' => ...
| inr h' => ...
```
The case where the subgoals are handled with `·` instead of `case` is more contentious (and much more numerous) so I left those alone. This pattern also appears with `cases'`, `induction`, `induction'`, and `rcases`. Furthermore, there is a similar transformation for `by_cases`:
```lean
by_cases h : cond
case pos => ...
case neg => ...
```
is replaced by:
```lean
if h : cond then
  ...
else
  ...
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alexjbest** [comment]
!bench

**Comment by alexjbest** [comment]
maintainer merge

**Comment by alexjbest** [comment]
Btw, to what extent was this automatically generated?

**Comment by digama0** [comment]
Not at all, although I did think about ways this kind of thing might be automated in the future.

File: Archive/Arithcc.lean
Comment (urkud) [inline Archive/Arithcc.lean:312]: Can we use `induction .. generalizing` instead of `revert; induction` here? Or you prefer to leave it to another PR?

Comment (digama0) [reply Archive/Arithcc.lean]: sure, we can remove all the `intro` as well.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
